### PR TITLE
write keys to file if path exists and file does not exist

### PR DESCRIPTION
**Pull request by GHadari**
- If the RSA keys path is specified, but no file exists in destination - new keys will be generated, and saved to the location indicated as path.
- The rest of the keys-selection behavior remains unchanged.
- keys variable check changed from `if x is None` to `if x` to include situation of empty string, as might appear in a config file.
- Added tests for keys creation

### CODE REVIEW

File: jwthenticator/tests/test_utils.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+@backup_environment
+@pytest.mark.asyncio
+# No keys or path inputted - create keys
+async def test_get_rsa_key_pair_no_input() -> None:
+    environ[PUBLIC_KEY_PATH_ENV_KEY] = ""
+    environ[PRIVATE_KEY_PATH_ENV_KEY] = ""
+    environ[PUBLIC_KEY_VALUE_ENV_KEY] = ""
+    environ[PRIVATE_KEY_VALUE_ENV_KEY] = ""
+    public_key, private_key = _reload_env_vars_get_rsa_key_pair()
+    assert public_key.count("PUBLIC KEY-----") == 2
```
Comment (tomer322) [inline jwthenticator/tests/test_utils.py:56]: Not sure what're we testing here, is it that the public/private keys are valid keys? If so, I think you can probably try to load them back to the library which created them

File: jwthenticator/utils.py [hunk 2]
```diff
@@ -14,31 +15,44 @@ def get_rsa_key_pair() -> Tuple[str, Optional[str]]:
...
+        return RSA_PUBLIC_KEY, RSA_PRIVATE_KEY
 
     return create_rsa_key_pair()
 
 
+def _read_rsa_keys_from_file() -> Tuple[str, Optional[str]]:
+    with open(RSA_PUBLIC_KEY_PATH, 'r', encoding='utf8') as f_obj:
+        public_key = f_obj.read()
+    private_key = None
+    if RSA_PRIVATE_KEY_PATH is not None:
```
Comment (tomer322) [inline jwthenticator/utils.py:41]: I know it was here before, but do you think this was intended? JWThenticator can be loaded with only a public key?

File: jwthenticator/tests/test_utils.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+import jwthenticator.utils
+from jwthenticator import consts
+from jwthenticator.utils import get_rsa_key_pair
+from jwthenticator.tests.utils import random_key, backup_environment
+
+PUBLIC_KEY_PATH_ENV_KEY = "RSA_PUBLIC_KEY_PATH"
+PRIVATE_KEY_PATH_ENV_KEY = "RSA_PRIVATE_KEY_PATH"
+PUBLIC_KEY_VALUE_ENV_KEY = "RSA_PUBLIC_KEY"
+PRIVATE_KEY_VALUE_ENV_KEY = "RSA_PRIVATE_KEY"
```
[the remark this answers is not in the document]
Comment (GHadari) [reply jwthenticator/tests/test_utils.py:15]: Had an issue overwriting them for the tests, plus this way the test is more extensive (goes over more of the flow).

File: jwthenticator/utils.py (hunk 2 above)
Comment (GHadari) [reply jwthenticator/utils.py:41]: I don't know, but this test was persistent across the function so I avoided changing the behavior. What's your opinion on this?

Comment (tomer322) [reply jwthenticator/utils.py:41]: So probably it was, not sure why though. You need the private key to both verify and sign tokens as far as I know.

**Pull request merged**